docs/error-codes.md

```diff
@@ -266,4 +472,154 @@ the error code. For example:
 
 #### `IL2049`: Unrecognized internal attribute 'attribute'
 
-- The internal attribute name 'attribute' being used in the xml is not supported by the linker, check the spelling and the supported internal attributes.
\ No newline at end of file
+- The internal attribute name 'attribute' being used in the xml is not supported by the linker, check the spelling and the supported internal attributes.
+
+#### `IL2050` Trim analysis: Making a generic type instantiation from 'type' which has 'DynamicallyAccessedMembersAttribute' on some of its generic parameters. ILLink currently doesn't analyze type values for generic parameters when making a generic type instantiation via 'System.Type.MakeGenericType'
+
+- Analysis of type parameters for `System.Type.MakeGenericType` is not yet implemented. If the open generic type `type` has `DynamicallyAccessedMembersAttribute` on any of its generic parameters, ILLink currently can't validate that the requirements are fulfilled by the calling method.  
+
+  ``` C#
+  class Lazy<[DynamicallyAccessedMembers(DynamicallyAccessedMemberType.PublicParameterlessConstructor)] T> 
+  {
+      // ...
+  }
+  
+  void TestMethod()
+  {
+      // IL2050 Trim analysis: Making a generic type instantiation from 'Lazy<T>' which has 
+      // 'DynamicallyAccessedMembersAttribute' on some of its generic parameters. ILLink currently doesn't analyze 
+      // type values for generic parameters when making a generic type instantiation via 'System.Type.MakeGenericType'
+      typeof(Lazy<>).MakeGenericType(new Type[] { typeof(TestType) });
+  }
+  ```
+
+  #### `IL2051` Trim analysis: Unrecognized type value used to called the method 'MakeGenericType'. It's not possible to guarantee the availability of requirements of such type.
+
+- If `System.Type.MakeGenericType` is called on an type which is not statically analyzable. If the actual type has generic parameters with `DynamicallyAccessedMembersAttribute` ILLink would be required to fulfill the requirements declared by those attributes, but since the ILLink doesn't know the type, it can't determine if such requirements exist.  
+
+  ``` C#
+  void TestMethod(Type type)
+  {
+      // IL2051 Trim analysis: Unrecognized value passed to the parameter 'typeArguments' of method 'System.Type.MakeGenericType'
```

Comment:
Should we copy the message here (it can go out of sync) ?

Reply:
I was trying to show how the message would look with specific references to code - since the message with parametrized pieces can be hard to understand.